utils/ensemble_tools.py

```diff
@@ -0,0 +1,166 @@
+""" A module to create input directories and encode
+configfurations for FabNESO
+"""
+
+import os
+import shutil
+import itertools
+import re
+from pathlib import Path
+import lxml.etree as ET
+
+
+def create_dir_tree(
+    sweep_path,
+    n_dirs,
+    destructive,
+    copy_dir,
+    edit_file,
+    parameter_to_scan,
+    scan_range,
+    outdir_prefix,
+):
+    """Create a directory tree in the sweep_path"""
+
+    copy_files = os.path.isdir(copy_dir)
+    if not copy_files:
+        print("No correct copy dir found, just creating the tree for now!")
+
+    # Make the base directory
+    try:
+        make_directory(sweep_path, destructive)
+    except FileExistsError:
+        return
```

Comment:
I would leave the exception unhandled rather than returning without doing anything as its not clear to user that something went wrong otherwise and the error message should be informative

```suggestion
    make_directory(sweep_path, destructive)
```